addon/components/frost-numeric-select.js

```diff
@@ -0,0 +1,183 @@
+import {PropTypes} from 'ember-prop-types'
+import computed, {readOnly} from 'ember-computed-decorators'
+const {isPresent} = Ember
+
+export default Component.extend({
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+  layout,
+
+  // == PropTypes =============================================================
+
+  propTypes: {    // options
+    align: PropTypes.string,
+    isHookEmbedded: PropTypes.bool,
+    receivedHook: PropTypes.string,
+    tabindex: PropTypes.number,
+    type: PropTypes.string,
+    isArrowVisible: PropTypes.bool,
+    isAllowInputNegative: PropTypes.bool,
+    isAllowUseDownButtonToNegative: PropTypes.bool,
+    showError: PropTypes.bool,
+    errorMessage: PropTypes.string,
+    class: PropTypes.string,
+
+    // Setting these as part of establishing an inital value
+    autocapitalize: PropTypes.string,
+    autofocus: PropTypes.bool,
+    autocorrect: PropTypes.string,
+    form: PropTypes.string,
+    maxlength: PropTypes.number,
+    placeholder: PropTypes.string,
+    required: PropTypes.bool,
+    selectionDirection: PropTypes.string,
+    spellcheck: PropTypes.bool,
+    value: PropTypes.string,
+    title: PropTypes.string
+
+    // state
+  },
+
+  getDefaultProps () {
+    return {
+      // options
+      align: 'left',
+      autocapitalize: 'off',
+      autocorrect: 'off',
+      autofocus: false,
+      isArrowVisible: false,
+      isAllowInputNegative: true,
+      isAllowUseDownButtonToNegative:false,
+      showError: false,
+      errorMessage: 'Cannot input invalid character',
+      isHookEmbedded: false,
+      required: false,
+      spellcheck: false,
+      tabindex: 0,
+      type: 'text'
+    }
+  },
+
+  // == Functions =============================================================
+
+  _isNumberKey (e) {
+    let decimalExist = this.get('value') && this.get('value').indexOf('.') >=0
+    let charCode = (e.which) ? e.which : e.keyCode
+
+    if(!this.get('isAllowInputNegative')){
+      if (charCode != 46 && charCode > 31
+        && (charCode < 48 || charCode > 57)) {
+          return false
+       } else if (charCode == 46 && decimalExist) {//check if user can enter decimal point
+         return false
+       }
+      return true
+    } else {
+      if (charCode == 45 && !this.get('value')) {//check if user can enter negative symbol
+        return true
+      } else if (charCode == 46 && decimalExist) {
+        return false
+      } else if (charCode != 46 && charCode > 31
+        && (charCode < 48 || charCode > 57))
+         return false
+
+      return true
+    }
+  },
+
+  //calculate how many digit a value has
+  _calculateDigit (value) {
+    return value.slice(value.indexOf('.')+1).length
+  },
+
+  //set calculated and properly formated value
+  _setCalculatedValue (value, numberOfDigit) {
+    if(this.$('input').val().indexOf('.') > -1) {
```

Comment:
This function is used for prevent floating point number precision in JS calculation. (e.g 1.22 - 1 will equals 0.199998, now it will be 1.22 - 1 = 0.22). I am not sure if it is required though.